Added development branch

vinaybajjuri58 opened the pull request.

purvasheth commented on public/index.html:
```diff
@@ -0,0 +1,49 @@
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>React App</title>
```
Change this to 'Learn Finance' or whatever name you decide for your video library

vinaybajjuri58 replied:
Hey Purva 
Firstly thanks for taking the time and reviewing the code.
I will modify the title index.html file.

purvasheth commented on src/App.js:
```diff
@@ -0,0 +1,21 @@
+
+function App() {
+  useEffect(() => {
+    document.title = "Learn Finance";
```
Curious as to why here. The page-specific titles can be put this way. The main title can be put directly in index.html.

vinaybajjuri58 replied:
Yeah got it 
removed the title modifier from App.js

purvasheth commented on src/Routes.js:
```diff
@@ -0,0 +1,22 @@
+      <Route path="/history" element={<History />} />
+      <Route path="/liked" element={<Liked />} />
+      <Route path="/playlists" element={<Playlists />} />
+      <Route path="/playlists" element={<Playlists />} />
```
I think this is extra

vinaybajjuri58 replied:
Yeah 
It is extra
I didn't see that 
I will remove it

purvasheth commented on src/Routes.js:
```diff
@@ -0,0 +1,22 @@
+      <Route path="/" element={<VideoList />} />
+      <Route path="/video/:videoId" element={<Video />} />
+      <Route path="/playlist/:playlistId" element={<PlayListVideos />} />
+    </Routes>
```
you can also add a not found route

vinaybajjuri58 replied:
I will add the page not found route.

purvasheth commented on src/Context/DataProvider.js:
```diff
@@ -0,0 +1,51 @@
+      {
+        id: 123,
+        name: "Watch Later",
+        videos: ["8egWflM1xok"],
```
Curious as to why watch later has a default value here

vinaybajjuri58 replied:
I was just trying to check the UI of playlists displaying parts like if a playlist has a video its image will be the image of the first video else it will be some random image. So I have added a default value.

purvasheth commented on src/Context/ReducerFunction.js:
```diff
@@ -0,0 +1,121 @@
+    return playlists;
+  } else {
+    playlist.videos.push(videoId);
+    return playlists.map((item) => (item.id === playlistId ? playlist : item));
```
This is okay. But if you want to run the loop once when adding a new video, you can perform a nested update.
```
return playlists.map((playlist)=>playlist.id === playlistId? {...playlist, videos:[...playlist.videos,videoId]}:playlist )
```

This will remove the extra find.  You can add the `includes` check as well. Use an `if else` condition instead for ternary here if you feel that would make the code more readable.

vinaybajjuri58 replied:
Yeah thanks for the code snippet suggestion. Ill change the code.

purvasheth commented on src/Components/ModalContent.js:
```diff
@@ -0,0 +1,134 @@
+                <button
+                  className="icon-button button-style"
+                  onClick={() => {
+                    setSnackBarContent(`Deleted playlist : ${playlist.name}`);
```
This can be extracted to a function in the same component similar to `playlistCheckBoxHandler`

vinaybajjuri58 replied:
Yeah okay 
It will do that.
It will make the code look cleaner I guess.

purvasheth commented on src/Components/Playlist.js:
```diff
@@ -0,0 +1,32 @@
+  };
+  const playlistImage =
+    playlist.videos.length > 0
+      ? getVideoImage(playlist.videos[0])
```
why not just playlist.videos[0].imageUrl here?

vinaybajjuri58 replied:
playlist image is not just a playlist.videos[0].imageUrl because If it doesn't contain any video It might result in an error. So I added a ternary to check if videos length is greater than 0.

purvasheth reviewed (commented):
Hi @vinaybajjuri58,
Well-written code. Liked how everything is well organized. UI looks clean too.
Left a few minor comments.

sjsouvik commented on src/Context/DataContext/Data.js:
```diff
@@ -0,0 +1,92 @@
+  //   avatarSrc:
+  //     "https://yt3.ggpht.com/ytc/AAUvwnjhWBsemL-_0qiUL8CBztox-E6ZmYcVYW-ZxEPQ=s88-c-k-c0x00ffffff-no-rj",
+  //   imageUrl:
+  //     "https://i.ytimg.com/vi/BTOHpoC-XsE/hqdefault.jpg?sqp=-oaymwEcCNACELwBSFXyq4qpAw4IARUAAIhCGAFwAcABBg==&rs=AOn4CLCf75LQlYbk9yMVakwshsjdJzvFog",
```
Try removing commented code whenever you're done with the implementation

sneha-adlakha commented:
code is looking very neat and clean.few suggestions from my side
-while navigating users playlists i have noticed few things like if we have more than one videos in playlist ,i was not able to see all at one time,i think either we have option to see all at once or in case you are showing single video at a time you can add prev video view button same as you added next button
-if you can add logout button it will be great.
-for like and dislike video you can avoid toast messages i think for add to playlist/remove from playlist it is ok but for like/dislike we usually don’t see toast messages.just a suggestion

sjsouvik commented on src/Pages/VideoPage.js:
```diff
@@ -0,0 +1,139 @@
+            <button
+              className="icon-button button-style"
+              onClick={() => {
+                if (isLoggedIn) {
```
extract this kind of click handler(having lines >= 3) to a separate function which will make the code cleaner

sjsouvik reviewed (changes requested):
- remove commented code once a feature implementation is done
- since you've added snackbar related states in context, toast for one for page will be visible in another page as well, if you visit quickly
- try to move wherever inline style is applied, to a css file, then import and use (navbar )
- add some style to show active route in FE
- add _redirects in public folder so that can prevent page not found error in netlify, maybe because of this 404 page is also not working, please check it 
- extract click handler which contains many lines of code to make the code cleaner

Aravindan07 commented on src/Context/DataContext/DataProvider.js:
```diff
@@ -0,0 +1,51 @@
+import { useContext, useReducer, useState } from "react";
+import { DataContext } from "./DataContext";
+import { ReducerFunction } from "./ReducerFunction";
```
Use camelCase word for named exports and Capitalized word for default exports.

Aravindan07 commented on src/Context/DataContext/ReducerFunction.js:
```diff
@@ -0,0 +1,121 @@
+  }
+}
+
+function removeFormHistory({ history, videoId }) {
```
I hope the name might be removeFromHistory

vinaybajjuri58 replied:
Yeah it's a spell mistake Ill change it

Aravindan07 commented on src/Pages/VideoList.js:
```diff
@@ -0,0 +1,33 @@
+        {state.videos.map((item) => (
+          <div
+            key={item.id}
+            onClick={() => {
+              if (isLoggedIn) {
+                dispatch({
+                  type: ActionTypes.ADD_TO_HISTORY,
+                  payload: item,
+                });
+              }
+            }}
```
OnClick handler can be moved to a separate function rather than inline handling. Tanay once said not to use inline onClick handler. Also, try to use the ternary operator rather than the old if-else conditions.

Aravindan07 commented on src/Pages/VideoPage.js:
```diff
@@ -0,0 +1,139 @@
+          <button
+            className="icon-button button-style"
+            onClick={() => {
+              if (isLoggedIn) {
+                if (inLikedVideos({ id: video.id, likedVideos: state.liked })) {
+                  setSnackBarContent(`Removed from Liked Videos`);
+                  showSnackBar();
+                  dispatch({
+                    type: ActionTypes.REMOVE_FROM_LIKED,
+                    payload: video,
+                  });
+                } else {
+                  setSnackBarContent("Video is not present in liked videos");
+                  showSnackBar();
+                }
+              } else {
+                setSnackBarContent("Please Login");
+                showSnackBar();
+              }
+            }}
```
Can be moved outside of inline handler

Aravindan07 reviewed (changes requested):
Hey, Vinay design is good I really liked the design. I have a few suggestions to improve your design.
1. sidebar is not fixed in your applications you have 9 videos I guess but if you have like 20+videos the user has to scroll up every time to choose to navigate to different pages like playlist page, liked videos page, etc.

2. When I am using your app I clicked on the like button then it shows a message you are not logged in. But I can't able to figure out how to log in like in the icon should I click to log in.Then I went to all the options in the sidebar and figured it out. It would be better if you add login/signup in the navbar or in the sidebar for easy accessibility.

3. Padding in the sidebar menu can be reduced it feels like too big when i hover on the menu items. Just a suggestion!

vinaybajjuri58 commented:
> 
> 
> code is looking very neat and clean.few suggestions from my side
> -while navigating users playlists i have noticed few things like if we have more than one videos in playlist ,i was not able to see all at one time,i think either we have option to see all at once or in case you are showing single video at a time you can add prev video view button same as you added next button
> -if you can add logout button it will be great.
> -for like and dislike video you can avoid toast messages i think for add to playlist/remove from playlist it is ok but for like/dislike we usually don’t see toast messages.just a suggestion

Hey sneha 
thanks for the suggestions 
In the playlist part I tried implenting similar to youtube but didint do the css properly as of now added a symbol
I have added the logout button

sjsouvik commented:
Approved all the changes that are requested by me. Good job! @vinaybajjuri58

vinaybajjuri58 commented:
> 
> 
> Approved all the changes that are requested by me. Good job! @vinaybajjuri58

Thanks for taking your time and reviewing the code
